File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
  var answer = [];

  const candidate = [];
  for (const s of score) {
    candidate.push(s);
    candidate.sort((a, b) => a - b);
    if (candidate.length <= k) {
      answer.push(candidate[0]);
    } else {
      candidate.shift();
      answer.push(candidate[0]);
    }
  }

  console.log(answer);
  return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
import heapq
from typing import List

def solution(k: int, score: List[int]) -> List[int]:
    hall: List[int] = []
    result: List[int] = []
    for value in score:
        heapq.heappush(hall, value)
        if len(hall) > k:
            heapq.heappop(hall)
        result.append(hall[0])
    return result
